packages/data/src/registry.js

```diff
@@ -47,6 +47,11 @@ function isObject( object ) {
 	return object !== null && typeof object === 'object';
 }
 
+function getStoreName( storeNameOrDescriptor ) {
+	return isObject( storeNameOrDescriptor )
+		? storeNameOrDescriptor.name
+		: storeNameOrDescriptor;
```

Comment:
Nit and probably unrelated to this PR, feel free to ignore.

Maybe it'd be easier to type to check for `string` instead?

```suggestion
	return typeof storeNameOrDescriptor === 'string'
		? storeNameOrDescriptor
		: storeNameOrDescriptor.name;
```

Reply:
Sounds like a good idea, implemented it :+1: One way it could backfire is when someone passes `null` and then the `.name` property access will crash. Previously we'd just use the `null` as a key into the stores map and simply find nothing. Hope nobody really does this.